src/Language/Pirouette/QuasiQuoter/Syntax.hs

```diff
@@ -241,24 +358,52 @@ parseTerm = label "Term" $ makeExprParser pAtom ops
 
     parseBottom :: Parser (Expr lang)
     parseBottom = do
-      try (symbol "bottom")
+      P.try (symbol "bottom")
       ty <- symbol "@" >> parseTypeAtom
       return (ExprBottom ty)
 
+    parseCase :: Parser (Expr lang)
+    parseCase = do
+      P.try (symbol "case")
+      ty <- symbol "@" >> parseTypeAtom
+      tyRes <- symbol "@" >> parseTypeAtom
```

Comment:
One particular thing that might make things a little easier is to shuffle some types around:
```
case @Bool xs @(List Int)
  { Nil -> False
  ; Cons _ _ -> True
  }
```

Now it's pretty evident what is the result type and there's hopefully less confusion since this would be congruent with if-statements

Reply:
Are you ok to put the second `@type` after the `of` keyword? I think of possible parsing issues: the value we match against patterns could end with a type application and it may be hard to read and parse.